lib/timex_error.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Timex.TimexError do
+  @moduledoc """
+  Used for errors encountered during string formatting.
```

Comment:
This is what `FormatError` is for, I think you can remove this and use `ArgumentError` where you've used `TimexError` in this PR, as that is more appropriate anyway. If you don't agree, we can discuss it further though

Reply:
You are right, ArgumentError will be more suitable, I'll fix this